sql/updates/README.md

```diff
@@ -58,3 +58,45 @@ Note that modfiles that contain no changes need not exist as a
 file. Transition modfiles must, however, be listed in the
 `CMakeLists.txt` file in the parent directory for an update script to
 be built for that version.
+
+## Extension downgrades
+
+To support downgrades to previous versions of the extension, it is
+necessary to execute CMake from a Git repository since the generation
+of a downgrade script requires access to the previous version files
+that are used to generate an update script. In addition, we only
+generate a downgrade script to the immediate preceeding version and
+not to any preceeding version.
+
+A downgrade file consists of:
+- A prolog that is retrieved from the target version.
+- A version-specific piece of code that exists on the source version.
+- An epilog that is retrieved from the target version.
+
+The prolog consists of the files mentioned in 
+
+The version-specific code is found in the file
+`<version>--<update_from_version>.sql`. Values for `version` and
+`update_from_version` can be found in `version.config` file in the
+root. The prolog and epilog is fetched from the tag `version` in the
+Git repository, which means that you need to have a Git repository to
+generate a downgrade file.
+
+For downgrades to work correctly, some rules need to be followed:
+
+1. If you add a new object in `latest-dev.sql`, you need to remove it
+   in the version-specific downgrade file. The `pre-update.sql` in the
+   previous version do not know about objects created in this version,
+   so they need to be dropped explicitly.
```

Comment:
The part about pre-update.sql seems confusing to me. Maybe something like "Object created in the current version should be dropped explicitly be editing the generated downgrade script" would be better, assuming I got the idea right, which I'm not sure about. All in all, I don't quite understand what exact action this text wants from me.

Reply:
No, the generated downgrade script should not be edited. I added some more details. Please check.